sdk/services/kwentaToken.ts

```diff
@@ -449,6 +449,89 @@ export default class KwentaTokenService {
 		return { claimableRewards, totalRewards };
 	}
 
+	public async getClaimableRewardsAll(
+		epochPeriod: number,
+		isOldDistributor: boolean = true,
+		isOp: boolean = false
+	) {
+		const {
+			MultipleMerkleDistributor,
+			MultipleMerkleDistributorPerpsV2,
+			MultipleMerkleDistributorOp,
+		} = this.sdk.context.multicallContracts;
+		const { walletAddress } = this.sdk.context;
+
+		if (
+			!MultipleMerkleDistributor ||
+			!MultipleMerkleDistributorPerpsV2 ||
+			!MultipleMerkleDistributorOp
+		) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const periods = Array.from(new Array(Number(epochPeriod)), (_, i) => i);
+		const adjustedPeriods = isOldDistributor
+			? periods.slice(0, TRADING_REWARDS_CUTOFF_EPOCH)
+			: periods.slice(TRADING_REWARDS_CUTOFF_EPOCH);
+
+		const fileNames = adjustedPeriods.map(
+			(i) =>
+				`trading-rewards-snapshots/${
+					this.sdk.context.networkId === 420 ? `goerli-` : ''
+				}epoch-${i}${isOp ? '-op' : ''}.json`
+		);
+
+		const responses: EpochData[] = await Promise.all(
+			fileNames.map(async (fileName, index) => {
+				const response = await client.get(fileName);
+				const period = isOldDistributor
+					? index >= 5
+						? index >= 10
+							? index + 2
+							: index + 1
+						: index
+					: index + TRADING_REWARDS_CUTOFF_EPOCH;
+				return { ...response.data, period };
+			})
+		);
+
+		const rewards = responses
+			.map((d) => {
+				const reward = d.claims[walletAddress];
+
+				if (reward) {
+					return [reward.index, walletAddress, reward.amount, reward.proof, d.period];
+				}
+
+				return null;
+			})
+			.filter((x): x is ClaimParams => !!x);
+
+		const claimed: boolean[] = await this.sdk.context.multicallProvider.all(
+			rewards.map((reward) =>
+				isOldDistributor
+					? MultipleMerkleDistributor.isClaimed(reward[0], reward[4])
+					: isOp
+					? MultipleMerkleDistributorOp.isClaimed(reward[0], reward[4])
+					: MultipleMerkleDistributorPerpsV2.isClaimed(reward[0], reward[4])
+			)
+		);
+
+		const { totalRewards, claimableRewards } = rewards.reduce(
+			(acc, next, i) => {
+				if (!claimed[i]) {
+					acc.claimableRewards.push(next);
+					acc.totalRewards = acc.totalRewards.add(wei(formatEther(next[2])));
```

Comment:
I think wei already knows how to handle BigNumber as an initialiser so should be no need for formatEther

Reply:
Thanks. `formatEther` has been removed.